components/Common/ProgressionSidebar/index.tsx

```diff
@@ -1,26 +1,47 @@
+import { useTranslations } from 'next-intl';
 import type { ComponentProps, FC } from 'react';
 
 import ProgressionSidebarGroup from '@/components/Common/ProgressionSidebar/ProgressionSidebarGroup';
-import WithSidebarSelect from '@/components/withSidebarSelect';
+import WithRouterSelect from '@/components/withRouterSelect';
+import { useClientContext } from '@/hooks/react-server';
 
 import styles from './index.module.css';
 
 type ProgressionSidebarProps = {
   groups: Array<ComponentProps<typeof ProgressionSidebarGroup>>;
 };
 
-const ProgressionSidebar: FC<ProgressionSidebarProps> = ({ groups }) => (
-  <nav className={styles.wrapper}>
-    {groups.map(({ groupName, items }) => (
-      <ProgressionSidebarGroup
-        key={groupName.toString()}
-        groupName={groupName}
-        items={items}
-      />
-    ))}
+const ProgressionSidebar: FC<ProgressionSidebarProps> = ({ groups }) => {
+  const t = useTranslations();
+  const { pathname } = useClientContext();
+
+  const selectItems = groups.map(({ items, groupName }) => ({
+    label: groupName,
+    items: items.map(({ label, link }) => ({ value: link, label })),
+  }));
+
+  const currentItem = selectItems
+    .map(item => item.items)
+    .flat()
+    .find(item => pathname === item.value);
 
-    <WithSidebarSelect groups={groups} />
-  </nav>
-);
+  return (
+    <nav className={styles.wrapper}>
+      {groups.map(({ groupName, items }) => (
+        <ProgressionSidebarGroup
+          key={groupName.toString()}
```

Comment:
```suggestion
          key={groupName}
```

Reply:
.toString() needs to be called, as the type of `groupName` is `FormattedMessage` which isn't necessarily a string.